prow/plank/controller.go

```diff
@@ -564,3 +567,25 @@ func getPodBuildID(pod *coreapi.Pod) string {
 	logrus.Warningf("BUILD_ID was not found in pod %q: streaming logs from deck will not work", pod.ObjectMeta.Name)
 	return ""
 }
+
+func (c *Controller) patchProwjob(srcPJ prowapi.ProwJob, destPJ prowapi.ProwJob) error {
+	srcPJData, err := json.Marshal(srcPJ)
+	if err != nil {
+		return fmt.Errorf("cannot create JSON patch: error marshal source prowjob: %v", err)
+	}
+
+	destPJData, err := json.Marshal(destPJ)
+	if err != nil {
+		return fmt.Errorf("cannot create JSON patch: error marshal dest prowjob: %v", err)
+	}
+
+	patch, err := jsonpatch.CreateMergePatch(srcPJData, destPJData)
+	if err != nil {
+		return fmt.Errorf("cannot create JSON patch: %v", err)
+	}
+
+	logrus.Infof("Created merge patch: %v", string(patch))
```

Comment:
This seems like an excessive amount of logging

Reply:
I think we don't include debug level logs now? If so I'll switch to debug.

Reply:
I think we always run at debug. Ask the question -- what does this help an admin answer? It would only be useful when the patch was successfully applied to the server but wrong in the way it effected the resource, right? We should have test coverage there.

Reply:
removed